### ISSUE TITLE
Reporting 'unknown value' despite being set

### ISSUE THREAD

**Issue by brandonh-msft** [terraform-linters/tflint#1112]
I'm getting an invalid error running tflint in a repo containing tf modules:

main.tf
```hcl
module "storage" {
  source        = "./storage"
  baseName      = var.baseName
  resourceGroup = azurerm_resource_group.rg
}
```

./storage/main.tf
```hcl
variable "baseName" { type = string }

variable "resourceGroup" {
  type = object({
    name     = string,
    location = string
  })
}

resource "azurerm_storage_account" "storage_account_raw_data" {
  name                     = "${var.baseName}st"
  resource_group_name      = var.resourceGroup.name
  location                 = var.resourceGroup.location
  account_tier             = "Standard"
  account_kind             = "StorageV2"
  access_tier              = "Hot"
  account_replication_type = "LRS"
}

```

.tflint.hcl
```hcl
config {
    module=true
    disabled_by_default = false

    variables = [ "baseName=lint" ]
}

plugin "azurerm" {
    enabled = true
}
```

Error reported from tflint (module=true):

~~~
6:17:59 runner_eval.go:123: [WARN] Unknown value found in storage/main.tf:12; Please use environment variables or tfvars to set the value; TFLint ignores an expression includes an unknown value.
~~~

pointing to
```hcl
  resource_group_name      = var.resourceGroup.name
```

which is clearly set in the module's usage.

## Version

```console
TFLint version 0.28.0
+ ruleset.azurerm (0.9.0)

Terraform v0.15.0
on linux_amd64
+ provider registry.terraform.io/hashicorp/azurerm v2.51.0
+ provider registry.terraform.io/hashicorp/random v3.1.0
```

**Comment by brandonh-msft** [comment]
Thinking tflint just has a problem with `object` variable types, I changed `./storage/main.tf` to

```hcl
variable "baseName" { type = string }

variable "resource_group_name" { type=string }

variable "location" { type=string }

resource "azurerm_storage_account" "storage_account_raw_data" {
  name                     = "${var.baseName}st"
  resource_group_name      = var.resource_group_name
  location                 = var.location
  account_tier             = "Standard"
  account_kind             = "StorageV2"
  access_tier              = "Hot"
  account_replication_type = "LRS"
}
```

Yet still receive the same error pointing to the `resource_group_name` line.

So then I ran `tflint --var='resource_group_name=foo'` but the error persists.

**Comment by bendrucker** [comment]
That's a warning log about TFLint being unable to evaluate that attribute. Is there an actual error here or is this just about seeing a warning message?

**Comment by bendrucker** [comment]
Doesn't matter whether you define the variable or not. TFLint cannot evaluate attributes referenced across resources, ever. Even if the referenced attribute itself is set to a a static value.

**Comment by brandonh-msft** [comment]
Additionally, the warning message could do with the clarification you're offering here instead of its current verbiage. As-is, it makes it sound like I could potentially do something to fix this, when I can't.

**Comment by bendrucker** [comment]
Sure, I can tune up the wording, but is there an actual problem here? Sounds like you turned on additional logs and are running into low level details of how TFLint works but there's no actual problem here.

**Comment by brandonh-msft** [comment]
I'm not sure what to say, there.

I would argue that throwing a warning for something that's unfixable isn't a valid thing to do to an end user. Think about compiler warnings in other languages; those are always fixable through some refactor/rewrite of the code. If you're intending to inform the user "hey, you have this thing here but I'm not in a position to actually do validation on it" then I wouldn't warn; I'd put it under debug/trace/verbose.

This also addresses your comment of a "low level detail" - I have the log level set to "info". If this is a low level detail, I shouldn't be seeing it at the info+ level, I should see it at a log level _lower_ than that.

**Comment by bendrucker** [comment]
Sure, I see the case for making this a different log level. The normal mode of operations is to not print any logs though. This is not a compiler where warnings are showing up default output.

It is inevitable that once you start turning on extra logging you may end up printing low level messages that seem like something is wrong. If I run `terraform init` I see:

```
TF_LOG=info terraform init
2021-04-30T11:41:32.398-0700 [INFO]  Terraform version: 0.15.1
2021-04-30T11:41:32.398-0700 [INFO]  Go runtime version: go1.16.2
2021-04-30T11:41:32.398-0700 [INFO]  CLI args: []string{"/usr/local/Cellar/tfenv/2.2.0/versions/0.15.1/terraform", "init"}
2021-04-30T11:41:32.398-0700 [INFO]  Loading CLI configuration from /Users/ben/.terraformrc
2021-04-30T11:41:32.399-0700 [WARN]  found legacy credentials "terraform-credentials-keychain"
2021-04-30T11:41:32.400-0700 [INFO]  CLI command args: []string{"init"}

Initializing the backend...
2021-04-30T11:41:33.912-0700 [INFO]  Failed to read plugin lock file .terraform/plugins/darwin_amd64/lock.json: open .terraform/plugins/darwin_amd64/lock.json: no such file or directory
```

The WARN's a mystery since that credentials helper is installed according to the docs (perhaps the plugin executable discovery code is warning on the lack of version suffix even though credentials helpers don't support that?

Then there's the INFO which is analogous to the case we're talking about. As the result of an expected error, Terraform is disabling a feature. I do agree it's the correct output but any error output at all is liable to generate anxiety (and issues) over what may be an entirely innocuous/normal condition.
